draft-wood-tls-external-psk-importer.md

```diff
@@ -124,23 +136,56 @@ used to derive concrete PSKs.
 
 # TLS 1.2 Compatibility
 
-Key importers do not affect TLS 1.2 in any way. Recall that TLS 1.2 permits computing the TLS PRF with
-any hash algorithm and PSK. Thus, a PSK may be used with the same KDF (and underlying HMAC hash algorithm) as
-TLS 1.3 with importers. However, critically, the derived PSK will not be the same since the importer
-differentiates the PSK via the identity and hash function. Thus, TLS 1.3 imported PSKs are distinct
-from those used in TLS 1.2 and avoid cross-protocol collisions.
+As specified, key importers do not affect TLS 1.2. Recall that TLS 1.2 permits computing the TLS PRF
+with any hash algorithm and PSK. Thus, an external PSK may be used with the same KDF (and underlying
+HMAC hash algorithm) as TLS 1.3 with importers. However, critically, the derived PSK will not
+be the same since the importer differentiates the PSK via the identity and hash function. Thus,
+TLS 1.3 imported PSKs are distinct from those used in TLS 1.2 and avoid cross-protocol collisions.
+
+However, as an added defense, keys imported for an endpoint that supports TLS 1.2 and 1.3 SHOULD
+derive a PSK specific to TLS 1.2 by using "tls12" as ImportedIdentity.label and SHA256 as
+ImportedIdentity.hash. The hint for this imported PSK is not modified.
```

Comment:
By "TLS 1.2" here, do you mean just TLS 1.2 or TLS 1.0 through 1.2? TLS "1.2" PSKs exist in all three. OpenSSL's implementation, at least, freely mixes them up.

Although I'm a little confused how this paragraph is meant to interact with "As specified, key importers do not affect TLS 1.2.". It seems like the document is specifying how to stuff this draft's notion of EPSK into TLS 1.2

Reply:
It's meant to refer to all previous versions, so I changed the title and removed some bits.